articles/10-tips-for-working-with-node.js/index.md

```diff
@@ -0,0 +1,295 @@
+
+Here's a list of some of the most useful libraries that will help ease the development of Node.js applications.
+
+-   [Express](http://expressjs.com/): is a web application framework for Node.js that handles robust APIs and Web servers.
+
+-   [Gulp](https://www.npmjs.com/package/gulp): helps to automate development workflow. It helps to automate simple tasks in the development of web applications.
+
+-   [Moment](https://www.npmjs.com/package/moment): is an insubstantial JavaScript date library for parsing, validating, manipulating, and formatting dates.
+
+-   [Async](https://caolan.github.io/async/v3/): is a utility module that provides powerful straightforward functions for working with asynchronous JavaScript.
+
+-   [Lodash](https://lodash.com/): makes JavaScript easier by taking out the irritation of working with arrays, numbers, objects, strings, etc.
+
+-   [JSHint](https://www.npmjs.com/package/jshint): detects errors and potential problems in your Node.js (JavaScript) code. You can easily adjust it in th environment you expect your code to execute.
+
+-   [Cheerio](https://cheerio.js.org/): Commonly used for web page scraping. It implements the subset of jQuery and removes `DOM`inconsistency and browser cleverness to extract API endpoint URLs from web pages.
+
+-   [Browserify](https://github.com/browserify/browserify): analyses `require ()` calls in your Node.js application to create a bundle that you can serve the browser in a single `<script>` tag.
+
+-   [Nodemailer](https://nodemailer.com/about/): Allows (easy as cake) email sending.
+
+You can have a look at more available libraries on [GitHub](https://github.com/aravindnc/A-to-Z-List-of-Useful-Node.js-Modules).
+
+## 3. The Node.js HTTP Server Module
+
+Node.js. has a built-in module called HTTP (Hypertext Transfer Protocol). It helps in making requests as a client and running the server to accept requests and return responses. 
+
+In this case, when you visit a webpage from a browser, you make a request to another computer (webserver) via the internet, which then provides you the webpage response. The webserver receives HTTP requests from the client, which
+listens to a server port and provides an HTTP response for example a HTML Page. 
+
+To use HTTP and client server in Node.js you must use `require ('http');`. Let's look at a simple Hello World HTTP Server module that will listen on port number 3000.
+```js
+var http = require('http')
+
+var server = http.createServer((function(request, response) {
+
+    response.writeHead(200, { "Content-Type\": "text/plain" });
+
+    response.end("Hello World\n");
+
+}));
+server.listen(3000);
+```
+When you type localhost:3000 on your browser, The output will be "Hello
```

Comment:
I would mention that readers would have to run `node index.js` in the terminal for the file to run. Beginners may not understand this.